storage/database/db_manager.go

```diff
@@ -1167,8 +1288,11 @@ func (dbm *databaseManager) WriteMerkleProof(key, value []byte) {
 
 // Cached Trie Node operation.
 func (dbm *databaseManager) ReadCachedTrieNode(hash common.Hash) ([]byte, error) {
+	dbm.lockInMigration.RLock()
+	defer dbm.lockInMigration.RUnlock()
+
 	if dbm.inMigration {
```

Comment:
You better use `dbm.InMigration()`.
This comment is for all places using `.inMigration`.

Reply:
This code is intended. 
This `RLock` can prevents  to access removed `MigrationDB`.